libs/plugins/flow-client/src/lib/flow.component.html

```diff
@@ -1,81 +1,56 @@
-          <flogo-logs></flogo-logs>
-        </div>
-        <div
-          class="flogo-flow-detail-menu"
-          [class.--active]="isflowMenuOpen"
-          (fgClickOutside)="closeFlowMenu()"
-        >
-          <i class="flogo-icon-kebabmenu kebabmenu-icon" (click)="toggleFlowMenu()"></i>
-          <div class="flogo-flow-detail-menu-list-wrapper">
-            <ul class="flogo-flow-detail-menu-list">
-              <!--
-                <li class="flogo-flow-detail-menu-item"><a disabled href="javascript:void(0);"><i class="fa fa-clone fa-fw"></i>{{ 'CANVAS:COPY' | translate }}</a></li>
-              -->
-              <li class="flogo-flow-detail-menu-item">
-                <a
-                  (click)="deleteFlow()"
-                  href="javascript:void(0);"
-                  class="flogo-flow-detail-menu-anchor"
-                >
-                  <i class="flogo-icon flogo-icon-delete"></i
-                  >{{ 'CANVAS:DELETE' | translate }}
-                </a>
-              </li>
-            </ul>
-          </div>
-        </div>
+  <flogo-designer-header
+    [name]="flowState.name"
+    [appName]="flowState?.app?.name"
+    [description]="flowState.description"
+    (goBack)="navigateToApp()"
+    (changeName)="changeFlowDetailName($event, 'name')"
+    (changeDescription)="changeFlowDetail($event, 'description')"
+  >
+    <svg class="flow-logo" viewBox="-3 -11 48 48" header-role="resource-logo">
+      <use xlink:href="/assets/svg/plugin-flows.svg#flowIcon"></use>
+    </svg>
+    <!--Temporarily removing run flow for v0.9.0-->
+    <!--<flogo-run-flow
+        header-role="right-block"
+        [disableRun]="runnableInfo?.disabled"
+        disableReason="{{ runnableInfo?.disableReason | translate }}"
+        [flowInputs]="flowState?.metadata?.input"
+        (submit)="onRunFlow($event)"
+      ></flogo-run-flow>-->
+    <flogo-logs header-role="right-block"></flogo-logs>
+    <div
+      header-role="right-block"
+      class="menu-dropdown"
+      (fgClickOutside)="closeFlowMenu()"
+    >
+      <i
+        class="flogo-icon-kebabmenu kebabmenu-icon menu-dropdown-more-button"
+        [class.active]="isflowMenuOpen"
+        (click)="toggleFlowMenu()"
+        role="button"
+        id="flowMenuButton"
+        aria-haspopup="true"
+        [attr.aria-expanded]="isflowMenuOpen"
+      ></i>
+      <div
+        class="menu-dropdown__content menu-dropdown__content--more"
+        aria-labelledby="flowMenuButton"
+        [class.active]="isflowMenuOpen"
+      >
+        <ul class="dropdown-list">
+          <li class="dropdown-list__item" (click)="deleteFlow()">
+            <i class="flogo-icon flogo-icon-delete"></i>{{ 'CANVAS:DELETE' | translate }}
+          </li>
+        </ul>
       </div>
-    </section>
-    <flogo-flow-tabs class="tabs"></flogo-flow-tabs>
-  </div>
+    </div>
+    <flogo-flow-tabs
+      header-role="bottom-block"
+      #bottomBlock
```

Comment:
I think #bottomBlock reference is used only to know whether bottom-block is need or not.
Can we query the bottom-block in component using the **header-role** attribute ? So can we can remove this local reference.
@ContentChild('[header-role=bottom-block]')

Reply:
As per my understanding on the selectors of ContentChild it works with Directive type / names. 

Please resolve the conversation if it helped.

Reply:
As mentioned above if we go for the directive approach where we declare a directive with selector `[header-role="bottom-block"]` then this could be encapsulated in the HeaderComponent.

If this sounds too complicated for the use case I think we can also take the following alternatives which are less elegant but far simpler:
1. Provide an option in the header to collapse the margin:
```html
<flogo-designer-header [collapseBottomBorder]=true></flogo-designer-header>
```

2. Don't do anything and let the user customize the border:
```ts
@Component({
   template: '<flogo-designer-header>...</flogo-designer-header>',
   styles: [`flogo-designer-header { padding-bottom: custom px }`]
})
class UsingHeaderComponent {
}

```

Reply:
Implemented the directive `BottomBlockDirective`